app/src/main/java/com/kickstarter/ui/activities/ProjectPageActivity.kt

```diff
@@ -304,75 +313,71 @@ class ProjectPageActivity :
-            .compose(bindToLifecycle())
             .observeOn(AndroidSchedulers.mainThread())
             .subscribe { showCreatePledgeSuccess(it) }
+            .addToDisposable(disposables)
 
         this.viewModel.outputs.projectMedia()
-            .compose(bindToLifecycle())
             .observeOn(AndroidSchedulers.mainThread())
             .subscribe { binding.mediaHeader.inputs.setProjectMedia(it) }
+            .addToDisposable(disposables)
 
         viewModel.outputs.playButtonIsVisible()
-            .compose(bindToLifecycle())
-            .compose(Transformers.observeForUI())
+            .observeOn(AndroidSchedulers.mainThread())
             .subscribe { binding.mediaHeader.inputs.setPlayButtonVisibility(it) }
+            .addToDisposable(disposables)
 
         viewModel.outputs.updateVideoCloseSeekPosition()
-            .compose(bindToLifecycle())
-            .compose(Transformers.observeForUI())
+            .observeOn(AndroidSchedulers.mainThread())
             .subscribe { binding.mediaHeader.inputs.setPlayerSeekPosition(it) }
+            .addToDisposable(disposables)
 
         binding.mediaHeader.outputs.onFullScreenClicked()
-            .compose(bindToLifecycle())
             .compose(Transformers.observeForUI())
             .subscribe { viewModel.inputs.fullScreenVideoButtonClicked(it) }
 
         binding.mediaHeader.outputs.playButtonClicks()
-            .compose(bindToLifecycle())
             .compose(Transformers.observeForUI())
             .subscribe { viewModel.inputs.onVideoPlayButtonClicked() }
 
         this.viewModel.outputs.onOpenVideoInFullScreen()
             .subscribeOn(Schedulers.io())
             .distinctUntilChanged()
-            .compose(bindToLifecycle())
-            .compose(Transformers.observeForUI())
+            .observeOn(AndroidSchedulers.mainThread())
             .subscribe {
                 this.startVideoActivity(startForResult, it.first, it.second)
-            }
+            }.addToDisposable(disposables)
 
         viewModel.outputs.backingViewGroupIsVisible()
-            .compose(bindToLifecycle())
-            .compose(Transformers.observeForUI())
+            .observeOn(AndroidSchedulers.mainThread())
             .subscribe {
                 binding.backingGroup.visibility = it.toVisibility()
-            }
+            }.addToDisposable(disposables)
 
         viewModel.outputs.hideVideoPlayer()
-            .compose(bindToLifecycle())
-            .compose(Transformers.observeForUI())
+            .observeOn(AndroidSchedulers.mainThread())
             .subscribe {
                 if (it) {
                     binding.projectAppBarLayout.setExpanded(false)
                 }
-            }
+            }.addToDisposable(disposables)
 
         binding.backIcon.setOnClickListener {
-            (this as BaseActivity<*>).back()
+            onBackPressedDispatcher.onBackPressed()
+            finishWithAnimation()
```

Comment:
This call can be removed, choose one, either call any of the callbacks direactly, or finish the animation. Asumming of course that the callback would be `finishWithAnimation()` like
```
 this.onBackPressedDispatcher.addCallback {
            finishWithAnimation()
        }
```